Deployment

gingin77 opened the pull request:
**What** Made updates needed to deploy to Ubuntu server configured with Nginx and Passenger using Capistrano.

gingin77 committed:
Add Capistrano, prepare db and secret config files to be in hidden

gingin77 committed:
Update gems and Capistrano files

gingin77 committed:
Set branch for deployment to deployment; set migration role to app

gingin77 committed:
Add cap-rails-collection gem to access rake db:seed for production

gingin77 committed:
Update Capfile; remove deploy/staging file

gingin77 committed:
Set deploy from branch to 'master'

zporter commented on Capfile:
```diff
@@ -0,0 +1,40 @@
+# require "capistrano/rvm"
+# require "capistrano/rbenv"
+# require "capistrano/chruby"
+# require "capistrano/bundler"
```
I'm curious if the `bundler` one is worth having too 🤔

gingin77 replied:
I've looked at the docs for capistrano/bundler but I'm not clear which scenarios I'm using right now that it could help with - https://github.com/capistrano/bundler

zporter replied:
Looks like it's automatically included by the capistrano-rails gem, so no need here.

zporter commented on Gemfile:
```diff
@@ -24,15 +22,28 @@ gem 'devise'
+  gem "capistrano", "~> 3.10", require: false
+  gem "capistrano-rails", "~> 1.3", require: false
+  gem 'capistrano-rails-collection'
+  gem 'capistrano-passenger', '~> 0.2.0'
```
`require: false` on these last 2 lines?

gingin77 replied:
Sure, makes sense. I looked up what `require: false` means in a gem file and found this post, which seemed very clear: https://stackoverflow.com/a/4800746/8481019

Is there anything else you'd add?

gingin77 replied:
👍

zporter replied:
that sums it up pretty well. I'll just add that you can see where gems are required [here](https://github.com/gingin77/hackernews_clone/blob/master/config/application.rb#L4-L6)

zporter commented on config/deploy.rb:
```diff
@@ -0,0 +1,43 @@
+
+# Default value for linked_dirs is []
+# append :linked_dirs, "log", "tmp/pids", "tmp/cache", "tmp/sockets", "public/system"
+append :linked_dirs, "log", "tmp/pids", "tmp/cache", "tmp/sockets", "vendor/bundle", "public/system", "public/uploads"
```
probably don't need `public/uploads` since this app doesn't deal with file uploads AFAIK

zporter replied:
same goes for `public/system`. I _think_ that's Paperclip's default dir.

zporter replied:
might be worth adding `node_modules` if that's a thing on this project

gingin77 replied:
Ok - that all makes sense, except I don't use node_modules at all right now so no need to add that. I don't have a `bundle` dir in the `vendor` dir, so I'm removing that one as well.

zporter replied:
Looks like the default install path for Bundler is `.bundle`, so that should be included in the `linked_dirs` here. Can read more about that over at the [`Capistrano::Bundler` project](https://github.com/capistrano/bundler), which is a dependency of capistrano-rails.

zporter reviewed (approved):
just a few minor comments / questions. looks great 👍

gingin77 committed:
Make changes related to gems used for deployment

Add Capistrano/bundler gem in Gemfilea and Capfile
Set require: false for all Cap gems in Gemfile
